src/System.Runtime/tests/System/TypeTests.cs

```diff
@@ -293,5 +293,29 @@ public static void ReflectionOnlyGetType()
             Assert.Throws<TypeLoadException>(() => Type.ReflectionOnlyGetType("", true, true));
             Assert.Throws<NotSupportedException>(() => Type.ReflectionOnlyGetType("System.Tests.TypeTests", false, true));
         }
+
+        private static IEnumerable<object[]> SZArrayOrNotTypes()
+        {
+            yield return new object[] {typeof(int[]), true};
+            yield return new object[] {typeof(string[]), true};
+            yield return new object[] {typeof(void), false};
+            yield return new object[] {typeof(int), false};
+            yield return new object[] {typeof(int[]).MakeByRefType(), false};
+            yield return new object[] {typeof(int[,]), false};
+            yield return new object[]{typeof(TypeTests), false };
+            yield return new object[] {Array.CreateInstance(typeof(int), new[] {2}, new[] {-1}).GetType(), false};
+            yield return new object[] {Array.CreateInstance(typeof(int), new[] {2}, new[] {1}).GetType(), false};
+            yield return new object[] {Array.CreateInstance(typeof(int), new[] {2}, new[] {0}).GetType(), true};
+            yield return new object[] {typeof(int[][]), true};
+            yield return new object[] { Type.GetType("System.Int32[]"), true };
+            yield return new object[] { Type.GetType("System.Int32[*]"), false };
```

Comment:
typeof(int).MakeArrayType() and typeof(int).MakeArrayType(1) are a less roundable way to get these types than calling Type.GetType().